Assets/MixedRealityToolkit/Interfaces/InputSystem/IMixedRealityRaycastProvider.cs

```diff
@@ -27,5 +28,11 @@ public interface IMixedRealityRaycastProvider : IMixedRealityService
         /// </summary>
         /// <returns>Whether or not the SphereCast hit something.</returns>
         bool SphereCast(RayStep step, float radius, LayerMask[] prioritizedLayerMasks, out MixedRealityRaycastHit hitInfo);
+
+        /// <summary>
+        /// Performs a graphics raycast against the specified layerMasks.
+        /// </summary>
+        /// <returns>The RaycastResult of the raycast.</returns>
+        RaycastResult GraphicsRaycast(EventSystem eventSystem, PointerEventData pointerEventData, LayerMask[] layerMasks);
```

Comment:
Would we want to separate this out into an `IMixedRealityGraphicsRaycastProvider` instead?

Reply:
What are you thoughts on the pros/cons of this? I was thinking about this and I can see a world in which maybe someone only wants to implement one of these, though I was hoping to reduce interface concept overload by saying:

"If you want to have custom raycast behavior, you should implement the raycast behavior interface." For folks who want to implement just one of those, you can always take the base class itself.

Reply:
That was mainly my thought, about only wanting to implement one. Worst case, they can copy over the existing implementation, which is just a single line anyway

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/d097f5b87365296b936b364da36630643e118fbd/Assets/MixedRealityToolkit.Services/InputSystem/DefaultRaycastProvider.cs#L39